packages/babel-plugin-transform-object-rest-spread/src/index.js

```diff
@@ -213,14 +264,20 @@ export default function({ types: t }) {
             );
           }
 
-          const [argument, callExpression] = createObjectSpread(
-            file,
-            path.node.left.properties,
-            ref,
-          );
+          const [
+            impureComputedPropertyDeclarators,
+            argument,
+            callExpression,
+          ] = createObjectSpread(leftPath, file, ref);
+
+          if (impureComputedPropertyDeclarators.length > 0) {
+            nodes.push(
+              t.variableDeclaration("var", impureComputedPropertyDeclarators),
+            );
+          }
 
           const nodeWithoutSpread = t.clone(path.node);
-          nodeWithoutSpread.right = ref;
+          nodeWithoutSpread.right = ref; //todo: shouldn't the if statement always execute for this to work
```

Comment:
I removed [the if condition](https://github.com/peey/babel/blob/26a3399e17bba349f842351ab0787916523f22a1/packages/babel-plugin-transform-object-rest-spread/src/index.js#L251). I couldn't figure out what it is there for, but my rationale for the `//todo` comment was that since `nodeWithoutSpread.right` can't be `undefined`, then it must be always executing.

All tests pass